api/caching/tasks.py

```diff
@@ -156,36 +156,41 @@ def update_storage_usage_with_size(payload):
         return
 
     action = payload['action']
+    provider = metadata.get('provider', 'osfstorage')
+
     target_file_id = metadata['path'].replace('/', '')
-    size = metadata.get('size', 0)
+    target_file_size = metadata.get('size', 0)
 
     current_usage = target_node.storage_usage or 0
     target_file = BaseFileNode.load(target_file_id)
 
-    if action in ['create', 'update']:
-        current_usage += size
-    elif action == 'delete':
-        size = target_file.versions.aggregate(Sum('size'))['size__sum']
-        if current_usage < size:
-            current_usage = 0
-        else:
-            current_usage -= target_file.versions.aggregate(Sum('size'))['size__sum']
-    elif action == 'move':
+    if target_file:
+
+        target_file_size = target_file.versions.aggregate(Sum('size'))['size__sum'] or target_file_size
```

Comment:
To clarify, in what cases will target_file be False-equivalent? And in what cases will the or not be short-circuited?

Reply:
In cases where the target file doesn't yet exist on OSFStorage, such as create. I updated the case statement to specify actions.